### ISSUE TITLE
Force one grapheme in the disjunction of string literals

### ISSUE THREAD

**Issue by ota-meshi** [ota-meshi/eslint-plugin-regexp#644]
**Motivation**

It may be useful to be able to force the disjunction of string literals to use only strings that humans recognize as single characters.

**Description**

It is possible to use strings of two or more characters in the disjunction of string literals, but normal disjunction is also possible. (However, there are differences in the evaluation order.)
Forcing a single character to be used in the disjunction of string literals allows people to distinguish between uses.

I think it is better to implement the rule by using `Intl.Segmenter` and allowing it if it is recognized as one grapheme.

**Examples**

```js
/* ✓ GOOD */
var foo = /[\p{RGI_Emoji}--\q{🇦🇨|🇦🇩|🇦🇪|🇦🇫|🇦🇬|🇦🇮|🇦🇱|🇦🇲|🇦🇴|🇦🇶|🇦🇷|🇦🇸|🇦🇹|🇦🇺|🇦🇼|🇦🇽|🇦🇿|🇧🇦|🇧🇧|🇧🇩|🇧🇪|🇧🇫|🇧🇬|🇧🇭|🇧🇮|🇧🇯}]/v

/* ✗ BAD */
var foo = /[\q{abc|def}]/v
```
